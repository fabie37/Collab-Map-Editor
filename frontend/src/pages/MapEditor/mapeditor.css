body.map_body {
    background: black;
}