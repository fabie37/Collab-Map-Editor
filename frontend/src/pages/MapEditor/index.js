import React, { useContext } from 'react';
import MapContainer from '../../components/MapCreator/MapContainer/MapContainer';
import { Redirect } from 'react-router-dom';
import { MapContext } from '../../context/MapState';
import './mapeditor.css';

export default function MapEditor() {
    const { workingMap } = useContext(MapContext);

    // If map not selected, redirect
    if (!workingMap) {
        return <Redirect to='/mymaps' />;
    }

    return (
        <>
            <body className='map_body'>
                <MapContainer></MapContainer>
            </body>
        </>
    );
}
