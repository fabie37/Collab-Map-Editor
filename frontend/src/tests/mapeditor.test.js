import React from 'react';
import renderer from 'react-test-renderer';
import MapEditor from '../pages/MapEditor';
//import Link​​ from 'react-router-dom/Link';
let BrowserRouter = require('react-router-dom').BrowserRouter;
let Link = require('react-router-dom').Link;
let Route = require('react-router-dom').Route;

test('MapEditor Page renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Route path="/mapeditor" component={MapEditor} />
      </BrowserRouter>
    )
    .toJSON();
  // put the existing a link here instead of creating a sample facebook one
  expect(tree).toMatchSnapshot();
});
