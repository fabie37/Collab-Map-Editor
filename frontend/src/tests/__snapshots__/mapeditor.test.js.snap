// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapEditor Page renders correctly 1`] = `null`;
